### ISSUE TITLE
Being teleported by an angel while being mounted on an entity causes weird side effects.

### ISSUE THREAD

**Issue by boot2big** [Swirtzly/Weeping-Angels#102]
Bug: If on an entity (boat, minecart, etc) and teleported by an angel, the player will be teleported, and instantly snap back to the entity they're on... only, if they're not looking at the angel after this happens, it will keep repeating until eventually the player is sent to another dimension, where they'll be brought back to that very entity in the original dimension you did this in, only slightly laggier each time you go to another dimension.
Testing this with pigs does something else though, rather if they're transported to another dimension, they _might_ be dismounted, but breaking blocks leaves no drops, placing blocks consumes the item, but does not place it in the world, all mobs/entities are frozen and chunks do not render past the chunks that were loaded after the player "dismounts", but still seem to load

Reproduction steps: Place down a boat/minecart/saddled pig, spawn an angel a bit of a distance away, sit in the spawned transportation method, and turn your back to the angel.

tldr sitting on a mob or minecart does not play well with angel teleportation

**Comment by Swirtzly** [comment]
Hey! Thanks for reporting!
This will be a really simple fix! I will probably have it fixed tonight, if
not, then by the end of this week
All I need to do is just have the player dismount whatever they're riding
and then be teleported
